ckan/views/feeds.py

```diff
@@ -0,0 +1,484 @@
+
+    alternate_url = _alternate_url(params)
+
+    guid = _create_atom_id(u'/feeds/dataset.atom')
+
+    desc = u'Recently created or updated datasets on %s' % SITE_TITLE
+
+    return output_feed(results, feed_title=SITE_TITLE,
+                       feed_description=desc,
+                       feed_link=alternate_url,
+                       feed_guid=guid,
+                       feed_url=feed_url,
+                       navigation_urls=navigation_urls)
+
+
+def custom():
+    """
+    Custom atom feed
+
+    """
+    q = request.params.get(u'q', u'')
+    fq = u''
+    search_params = {}
+    for (param, value) in request.params.items():
+        if param not in [u'q', u'page', u'sort'] \
+                and len(value) and not param.startswith(u'_'):
+            search_params[param] = value
+            fq += u'%s:"%s"' % (param, value)
+
+    page = h.get_page_number(request.params)
+
+    limit = ITEMS_LIMIT
+    data_dict = {
+        u'q': q,
+        u'fq': fq,
+        u'start': (page - 1) * limit,
+        u'rows': limit,
+        u'sort': request.params.get(u'sort', None)
+    }
+
+    item_count, results = _package_search(data_dict)
+
+    navigation_urls = _navigation_urls(request.params, item_count=item_count,
+                                       limit=data_dict['rows'],
+                                       controller=u'feeds',
+                                       action=u'custom')
+
+    feed_url = _feed_url(request.params, controller=u'feeds', action=u'custom')
+
+    atom_url = h._url_with_params(u'/feeds/custom.atom', search_params.items())
+
+    alternate_url = _alternate_url(request.params)
+
+    site_title = config.get(u'ckan.site_title', u'CKAN')
+
+    return output_feed(results,
+                       feed_title=u'%s - Custom query' % site_title,
+                       feed_description=u'Recently created or updated'
+                       ' datasets on %s. Custom query: \'%s\'' % (site_title,
+                                                                  q),
+                       feed_link=alternate_url,
+                       feed_guid=_create_atom_id(atom_url),
+                       feed_url=feed_url,
+                       navigation_urls=navigation_urls)
+
+
+def _alternate_url(params, **kwargs):
+    search_params = params.copy()
+    search_params.update(kwargs)
+
+    # Can't count on the page sizes being the same on the search results
+    # view.  So provide an alternate link to the first page, regardless
+    # of the page we're looking at in the feed.
+    search_params.pop(u'page', None)
+    return _feed_url(search_params, controller=u'package', action=u'search')
+
+
+def _feed_url(query, controller, action, **kwargs):
+    """
+    Constructs the url for the given action.  Encoding the query
+    parameters.
+    """
+    # endpoint = controller + '.' + action
+    path = h.url_for(controller=controller, action=action, **kwargs)
```

Comment:
Would be good to refactor this to use the new Flask based `url_for` style. Also I'm pretty sure that if we pass the query params as `kwargs` to `url_for` they will get encoded and we don't need to the call below to `_url_with_params`

Reply:
donce